pkg/build/graph/analysis/bc.go

```diff
@@ -68,6 +76,60 @@ bc:
 	return markers
 }
 
+// FindMissingInputImageStreams checks all build configs and confirms that their From element exists
+//
+// Precedence of failures:
+// 1. A build config's input points to an image stream that does not exist
+// 2. A build config's input uses an image stream tag reference in an existing image stream, but no images within the image stream have that tag assigned
+// 3. A build config's input uses an image stream image reference in an exisiting image stream, but no images within the image stream have the supplied image hexadecimal ID
+func FindMissingInputImageStreams(g osgraph.Graph) []osgraph.Marker {
+	markers := []osgraph.Marker{}
+
+	for _, bcNode := range g.NodesByKind(buildgraph.BuildConfigNodeKind) {
+		for _, istNode := range g.PredecessorNodesByEdgeKind(bcNode, buildedges.BuildInputImageEdgeKind) {
```

Comment:
yep, name could be better (either IST or ISI).  The for loop of the other type was just bypassed, but the switch case does make it more explicit / clearer.  Introduced as part of next push.

As an FYI, I tried to leverage the Found() method on IST (and temporarily added one to ISI) to bypass the Successor call, but IsFound is not getting set on those nodes.  Only the imageStreaNode.Found() returned with the correct value.  My initial take was to not deal with this as part of this pull.